thrust/system/cuda/detail/scan.h

```diff
@@ -26,762 +26,204 @@
  ******************************************************************************/
 #pragma once
 
-
-#if THRUST_DEVICE_COMPILER == THRUST_DEVICE_COMPILER_NVCC
```

Comment:
Let's leave this in for now. @jaredhoberock @griwes does anyone know the historical reason why we had to wrap the CUDA backend headers in `#if THRUST_DEVICE_COMPILER == THRUST_DEVICE_COMPILER_NVCC`?

Reply:
Done.